src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(`${oldSeparator}`);
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateObject.YY) {
+    const year = Number(dateObject.YY);
+
+    dateObject.YYYY = year < 30 ? '20' + dateObject.YY : '19' + dateObject.YY;
+  }
+
+  if (dateObject.YYYY) {
+    dateObject.YY = dateObject.YYYY.slice(-2);
+  }
```

Comment:
Similarly, when slicing the 'YYYY' format to 'YY', you should also check if 'YY' is included in the 'toFormat' before performing this operation. This will prevent unnecessary processing and ensure that the conversion is only done when required by the 'toFormat'.